File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }
        int a = 0;
        boolean flag = false;
        for (int i : freq) {
            if (i != 0 && !flag) {
                a = i;
                flag = true;
            } 
            if (i != 0 && a != i) return false;
        }
        return true;
    }
}

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.js
/**
 * @param {string} s
 * @return {boolean}
 */
var areOccurrencesEqual = function(s) {
    const freq = new Array(26).fill(0);
    for (const c of s) 
        freq[c.charCodeAt(0) - 97]++;
        
    let a = 0;
    for (const i of freq) {
        if (i === 0) continue;
        if (a === 0) a = i;
        if (a !== i) return false;
    }
    return true;
};